src/Microsoft.ML.Data/Transforms/TermTransform.cs

```diff
@@ -144,6 +147,41 @@ public sealed class Arguments : ArgumentsBase
             public Column[] Column;
         }
 
+        internal sealed class ColInfo
+        {
+            public readonly string Name;
+            public readonly string Source;
+            public readonly ColumnType TypeSrc;
+
+            public ColInfo(string name, string source, ColumnType type)
+            {
+                Name = name;
+                Source = source;
+                TypeSrc = type;
+            }
+        }
+
+        public class ColumnInfo
+        {
+            public ColumnInfo(string input, string output, int maxNumTerms = Defaults.MaxNumTerms, SortOrder sort = Defaults.Sort, string[] term = null, string terms = null, bool textKeyValues = false)
```

Comment:
>string terms = null [](start = 153, length = 19)

Certainly don't need both of these.

Reply:
Considering how TermMap class and train function implemented...
I can spend probably few hours and refactor it, but this PR already covering so much stuff.

---
In reply to: [214463561](https://github.com/dotnet/machinelearning/pull/759#discussion_r214463561) [](ancestors = 214463561)

Reply:
One easy thing we can do is to pass `terms = null` everywhere, without allowing user to override it

---
In reply to: [214469361](https://github.com/dotnet/machinelearning/pull/759#discussion_r214469361) [](ancestors = 214469361,214463561)

Reply:
Right. In the public facing method, we definitely do not want both exposed. If an internal method requires it (why would you, if `terms` only possible use is the command line?), that's fine? I guess?

---
In reply to: [214479433](https://github.com/dotnet/machinelearning/pull/759#discussion_r214479433) [](ancestors = 214479433,214469361,214463561)

Reply:
Is this better?

---
In reply to: [214481635](https://github.com/dotnet/machinelearning/pull/759#discussion_r214481635) [](ancestors = 214481635,214479433,214469361,214463561)